### PR TITLE
Create skeleton for application

### PR DESCRIPTION
**Pull request by SStegniy**
Basic changes of skeleton

### CODE REVIEW

File: figma-shop/product-json.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
+{
```
Comment (iyarymovych-reenbit) [inline figma-shop/product-json.ts:1]: For now this is okay that you have such `json` in your project, but it has to be on firebase DB.

File: figma-shop/src/app/app-routing.module.ts [hunk 2]
```diff
@@ -1,8 +1,22 @@
...
+import { ProductsComponent } from './pages/products/products.component';
 
-const routes: Routes = [];
+
+const routes: Routes = [
+  { path: '', redirectTo: 'products', pathMatch: 'full' },
+  { path: 'chat', component: ChatComponent },
+  { path: 'blog', component: BlogComponent },
+  { path: 'about', component: AboutComponent },
+  { path: 'careers', component: CareersComponent },
```
Comment (iyarymovych-reenbit) [inline figma-shop/src/app/app-routing.module.ts:16]: You have duplicate `careers` routes. Remove one.

File: figma-shop/src/app/app-routing.module.ts [hunk 3]
```diff
@@ -1,8 +1,22 @@
 import { NgModule } from '@angular/core';
```
Comment (iyarymovych-reenbit) [inline figma-shop/src/app/app-routing.module.ts:1]: Could you please make sure that all TS Lint issues are fixed in your project?
Could you please rename PR on some normal title (it should be taken from a branch name)?
`Create skeleton for application`

File: figma-shop/src/app/components/header/header-nav/header-nav.component.scss [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    .item {
+        display: inline-block;
+        padding: 5px 10px;
+
+        img {
+            display: inline;
+            margin-left: 5px;
+        }
+    }
+}
```
Comment (iyarymovych-reenbit) [inline figma-shop/src/app/components/header/header-nav/header-nav.component.scss:35]: GitHub needs empty line at the end of each file you want to add.

File: figma-shop/src/app/pages/about/about.component.spec.ts [hunk 5]
```diff
@@ -1,20 +1,20 @@
 import { async, ComponentFixture, TestBed } from '@angular/core/testing';
```
Comment (iyarymovych-reenbit) [inline figma-shop/src/app/pages/about/about.component.spec.ts:1]: Remove all specs from your project - we will not write any unit tests.
Generate components with skipping them: `ng g c component-name --skip-tests`

File: figma-shop/src/app/pages/product-details/product-details.component.ts [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+import { Product } from 'src/app/shared/models/product.model';
+import { ProductServiceService } from '../../shared/services/product-service.service';
+
+@Component({
+  selector: 'app-product-details',
+  templateUrl: './product-details.component.html',
+  styleUrls: ['./product-details.component.scss']
+})
+export class ProductDetailsComponent implements OnInit {
+  @Input() product: any;
```
Comment (iyarymovych-reenbit) [inline figma-shop/src/app/pages/product-details/product-details.component.ts:11]: Please, use interfaces for your entities in a project. Avoid `any`.

Comment (iyarymovych-reenbit) [reply figma-shop/src/app/pages/product-details/product-details.component.ts:11]: You can write them via interfaces, like this:
```
export interface ConnectionUseCaseInterface {
  subject: string;
  url: string;
  beginOfCheckTime: string;
  frequency: number;
  validity: number;
}

Comment (iyarymovych-reenbit) [reply figma-shop/src/app/pages/product-details/product-details.component.ts:11]: Or via classes' models, like this:
```
export class CostModel {
  public id: number;
  public costRate: string;
  public servePosition: ServePositionEnum;
  public commodity: CommodityEnum;
  public type: TypeEnum;
  public calculationType: CalculationTypeEnum;
  public comments: string;
  public categoryFromTo: CategoryFromToEnum;
  public categories: CostCategoryModel[];
}

File: figma-shop/src/app/pages/product-details/product-details.component.ts [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+  constructor(private productService: ProductServiceService) { }
+
+  ngOnInit(): void {
+  }
+
+  updateActive(isActive: boolean) {
```
Comment (iyarymovych-reenbit) [inline figma-shop/src/app/pages/product-details/product-details.component.ts:17]: Always write modificators for methods and fields in your classes:
image

File: figma-shop/src/app/shared/models/category.model.ts [hunk 8]
```diff
@@ -0,0 +1,6 @@
+import { ICategory } from '../interfaces/category.interface';
+
+export class Category implements ICategory {
+    constructor(public id: number,
```
Comment (iyarymovych-reenbit) [inline figma-shop/src/app/shared/models/category.model.ts:4]: Bad indentation.
```
constructor(
    public id: number,
    public name: string
) {}

File: figma-shop/src/app/shared/services/product-service.service.ts [hunk 9]
```diff
@@ -0,0 +1,38 @@
...
+  }
+ 
+  getdbProducts(): AngularFireList<any> {
+    return this.productRef;
+  }
+ 
+  deleteAll(): Promise<void> {
+    return this.productRef.remove();
+  }
+
```
Comment (iyarymovych-reenbit) [inline figma-shop/src/app/shared/services/product-service.service.ts:36]: Remove unneeded empty lines.
Try to have everything beautiful.

File: figma-shop/src/app/pages/product-details/product-details.component.ts (hunk 6 above)
Comment (iyarymovych-reenbit) [reply figma-shop/src/app/pages/product-details/product-details.component.ts:11]: But for skeleton you can skip this comment.

File: figma-shop/product-json.ts (hunk 1 above)
Comment (SStegniy) [reply figma-shop/product-json.ts]: Already deleted this file, it was only as an example

File: figma-shop/src/app/app-routing.module.ts (hunk 2 above)
Comment (SStegniy) [reply figma-shop/src/app/app-routing.module.ts]: Fix it, thank you

File: figma-shop/src/app/app-routing.module.ts (hunk 3 above)
Comment (SStegniy) [reply figma-shop/src/app/app-routing.module.ts]: Configured TSLint

File: figma-shop/src/app/components/header/header-nav/header-nav.component.scss (hunk 4 above)
Comment (SStegniy) [reply figma-shop/src/app/components/header/header-nav/header-nav.component.scss]: In this case TS Lint was helpfull

File: figma-shop/src/app/pages/about/about.component.spec.ts (hunk 5 above)
Comment (SStegniy) [reply figma-shop/src/app/pages/about/about.component.spec.ts]: All specs have been deleted

**Review by iyarymovych-reenbit — APPROVED** [review]

File: figma-shop/src/app/components/header/header-nav/header-nav.component.html [hunk 10]
```diff
@@ -1 +1,41 @@
...
+        <nav class="nav">
+            <ul class="list">
+                <li class="item">Electronics<img src="../../../../assets/images/arrow-down.svg" alt="dropdown"></li>
+                <li class="item">Food<img src="../../../../assets/images/arrow-down.svg" alt="dropdown"></li>
+                <li class="item">Clothes<img src="../../../../assets/images/arrow-down.svg" alt="dropdown"></li>
+                <li class="item">Skin<img src="../../../../assets/images/arrow-down.svg" alt="dropdown"></li>
+                <li class="item">Toys<img src="../../../../assets/images/arrow-down.svg" alt="dropdown"></li>
+                <li class="item">Special<img src="../../../../assets/images/arrow-down.svg" alt="dropdown"></li>
+                <li class="item">Sport<img src="../../../../assets/images/arrow-down.svg" alt="dropdown"></li>
+                <li class="item">Books<img src="../../../../assets/images/arrow-down.svg" alt="dropdown"></li>
```
Comment (Maksicheg) [inline figma-shop/src/app/components/header/header-nav/header-nav.component.html:32]: Find the way to define path to your assets like `"assets/images/... svg"`, not like `../../../... assets ...`

File: figma-shop/src/app/pages/product-details/product-details.component.html [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+      <label>First Name: </label> {{product[0].id}}
+    </div>
+    <div>
+      <label>Age: </label> {{product[0].title}}
+    </div>
+    <div>
+      <label>Active: </label> {{product[0].country}}
+    </div>
+   
+    <!-- <span class="button is-small btn-primary" *ngIf='product.active' (click)='updateActive(false)'>Inactive</span>
```
Comment (Maksicheg) [inline figma-shop/src/app/pages/product-details/product-details.component.html:13]: Remove all your commented code in pr

File: figma-shop/src/app/pages/products/products.component.html [hunk 12]
```diff
@@ -0,0 +1,36 @@
...
+            </div>
+            <div class="products col-md-9 col-lg-9 border">
+                <div class="product-card" *ngFor="let product of allProducts">
+                    <app-product-card [product]="product"></app-product-card>
+                </div>
+            </div>
+        </div>
+    </main>
+
+
```
Comment (Maksicheg) [inline figma-shop/src/app/pages/products/products.component.html:31]: You have to accustom yourself to write "clean" code, especially when you are making prs. Look to every space and aligning. In current case you have too many spaces in html and also at the end of the file should be only one space!

File: figma-shop/src/app/pages/products/products.component.ts [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+import { ProductServiceService } from '../../shared/services/product-service.service';
+
+
+@Component({
+  selector: 'app-products',
+  templateUrl: './products.component.html',
+  styleUrls: ['./products.component.scss']
+})
+export class ProductsComponent implements OnInit {
+  public allProducts: Array<IProduct>;
```
Comment (Maksicheg) [inline figma-shop/src/app/pages/products/products.component.ts:13]: For your future projects, writing `I` before interfaces that's an outdated practice... 
At our projects we always write as `ProductInterface` or `ProductModel`, but in your project this is up to you, of course.

File: figma-shop/src/app/shared/models/product.model.ts [hunk 14]
```diff
@@ -0,0 +1,25 @@
...
+        public country: string,
+        public color: string,
+        public size: string,
+        public buyProperties: string,
+        public stock: number | string,
+        public image: string,
+        public description: Array<IMessage>,
+        public reviews: Array<IMessage>,
+        public questions: Array<IMessage>){}
+}
```
Comment (Maksicheg) [inline figma-shop/src/app/shared/models/product.model.ts:25]: If you want to define `model` you just need to create it like this: 
```ts
export class ProductModel {
      public id: number;
      public category: CategoryInterface;
      public title: string;
      public price: number;
      public previousPrice: number;
      public freshness: string;
      public farm: string;
      public deliveryArea: string;
      public deliveryTime: number;
      public country: string;
      public color: string;
      public size: string;
      public buyProperties: string;
      public stock: number | string;
      public image: string;
      public description: MessageInterface[];
      public reviews: MessageInterface[];
      public questions: MessageInterface[];
}

**Review by Maksicheg — APPROVED** [review]

**Pull request merged**